backend/lucene/src/main/java/org/hibernate/search/backend/lucene/lowlevel/reader/impl/IndexReaderMetadataResolver.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.lucene.lowlevel.reader.impl;
+
+import java.util.Map;
+
+import org.hibernate.search.util.common.AssertionFailure;
+
+import org.apache.lucene.index.DirectoryReader;
+import org.apache.lucene.index.IndexReaderContext;
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.store.Directory;
+
+public final class IndexReaderMetadataResolver {
+
+	private final Map<DirectoryReader, String> indexNameByDirectoryReader;
+
+	IndexReaderMetadataResolver(Map<DirectoryReader, String> indexNameByDirectoryReader) {
+		this.indexNameByDirectoryReader = indexNameByDirectoryReader;
```

Comment:
Maybe we can wrap with `Collections#unmodifiableMap()`

Reply:
Doesn't seem useful here, since the map is not exposed publicly.

If the goal was to prevent the map from being modified by the caller, then we'd have to copy the map. But in this case I'm not worried: the caller can be trusted.